templates/task.yaml

```diff
@@ -0,0 +1,194 @@
+      description: |
+        Initialize and fetch git submodules.
+    - name: DEPTH
+      type: string
+      default: "1"
+      description: |
+        Perform a shallow clone, fetching only the most recent N commits.
+    - name: SSL_VERIFY
+      type: string
+      default: "true"
+      description: |
+        Set the `http.sslVerify` global git config. Setting this to `false` is
+        not advised unless you are sure that you trust your git remote.
+    - name: CRT_FILENAME
+      type: string
+      default: ca-bundle.crt
+      description: |
+        File name of mounted crt using ssl-ca-directory Workspace. Default value
+        is "ca-bundle.crt".
+    - name: SUBDIRECTORY
+      type: string
+      default: ""
+      description: |
+        Subdirectory relative to the `output` Workspace to clone the repository.
+    - name: SPARSE_CHECKOUT_DIRECTORIES
+      type: string
+      default: ""
+      description: |
+        Define the directory patterns to match or exclude when performing a
+        sparse checkout.
+    - name: DELETE_EXISTING
+      type: string
+      default: "true"
+      description: |
+        Clean out the contents of the destination directory if it already exists
+        before cloning.
+    - name: HTTP_PROXY
+      type: string
+      default: ""
+      description: |
+        HTTP proxy server for non-SSL requests.
+    - name: HTTPS_PROXY
+      type: string
+      default: ""
+      description: |
+        HTTPS proxy server for SSL requests.
+    - name: NO_PROXY
+      type: string
+      default: ""
+      description: |
+        Opt out of proxying HTTP/HTTPS requests.
+    - name: VERBOSE
+      type: string
+      default: "false"
+      description: |
+        Log the commands executed.
+    - name: GIT_INIT_IMAGE
+      type: string
+      default: {{ .Values.images.gitInit }}
+      description: |
+        The image providing the git-init binary that this Task runs.
+    - name: USER_HOME
+      type: string
+      default: "/home/git"
+      description: |
+        Absolute path to the user's home directory.
+
+  results:
+    - name: COMMIT
+      description: |
+        The precise commit SHA that was fetched by this Task.
+    - name: URL
+      description: |
+        The precise URL that was fetched by this Task.
+    - name: COMMITTER_DATE
+      description: |
+        The epoch timestamp of the commit that was fetched by this Task.
+
+  volumes:
+    - name: user-home
+      emptyDir: {}
+    - name: scripts-dir
+      emptyDir: {}
+
+  stepTemplate:
+    env:
+{{- include "environment-variables" . | nindent 6 }}
+{{- if .Values.stepTemplate.resources }}
+    resources:
+  {{- .Values.stepTemplate.resources | toYaml | nindent 6 }}
+{{- end }}
+{{- if .Values.stepTemplate.securityContext }}
+    securityContext:
+  {{- .Values.stepTemplate.securityContext | toYaml | nindent 6 }}
+{{- end }}
+
+  steps:
+    - name: load-scripts
+      image: $(params.GIT_INIT_IMAGE)
```

Comment:
Note that in https://issues.redhat.com/browse/TKNECO-33 we want to explicitely remove this 🙃. One reason is.. support 😅 If we allow user to use whatever images they want, it's nice for them, but **impossible** for us to support them.